src/main/java/org/jabref/gui/preferences/ExternalTabView.java

```diff
@@ -0,0 +1,140 @@
+package org.jabref.gui.preferences;
+
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.RadioButton;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.push.PushToApplication;
+import org.jabref.gui.util.ViewModelListCellFactory;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.JabRefPreferences;
+
+import com.airhacks.afterburner.views.ViewLoader;
+
+public class ExternalTabView extends VBox implements PrefsTab {
+
+    @FXML private TextField eMailReferenceSubject;
+    @FXML private CheckBox autoOpenAttachedFolders;
+
+    @FXML private ComboBox<PushToApplication> pushToApplicationCombo;
+    @FXML private TextField citeCommand;
+
+    @FXML private RadioButton useTerminalDefault;
+    @FXML private RadioButton useTerminalSpecial;
+    @FXML private TextField useTerminalCommand;
+
+    @FXML private RadioButton usePDFAcrobat;
+    @FXML private TextField usePDFAcrobatCommand;
+    @FXML private RadioButton usePDFSumatra;
+    @FXML private TextField usePDFSumatraCommand;
+
+    @FXML private RadioButton useFileBrowserDefault;
+    @FXML private RadioButton useFileBrowserSpecial;
+    @FXML private TextField useFileBrowserSpecialCommand;
+
+    private final DialogService dialogService;
+    private final JabRefPreferences preferences;
+    private final JabRefFrame frame;
+
+    private ExternalTabViewModel viewModel;
+
+    public ExternalTabView(DialogService dialogService, JabRefPreferences preferences, JabRefFrame frame) {
```

Comment:
not sure for root components, but You can Inject DialogService and could also inject the PreferencesService interface. You would then just need the getters/setter methods for each value

Reply:
Is there some way also to inject the the JabRefFrame?
This @Inject-DI-stuff is really black-magic-vodoo to me.

Reply:
It is not possible to inject the `JabRefFrame` (and we will also not add this functionality as we try to get rid of the omnipresence of `JabrefFrame`). Here you find an overview of the possible `@Inject` classes:
https://github.com/JabRef/jabref/blob/47e3cc5617e456df03ef8b48500c824d09cec004/src/main/java/org/jabref/gui/DefaultInjector.java#L31-L51 

If you want to dig deeper into how the injection works, have a look here: https://github.com/JabRef/afterburner.fx/tree/master/src/main/java/com/airhacks/afterburner/injection
But yes it is kind of black magic by design, since you normally you don't want to think about how the injection works.

Reply:
I think I somehow get an idea of how to use it.
If I inject the PreferencesService, would that imply, that I have to add getters and setters for every value accessed by the PreferenceDialog in PreferencesService (which is basically almost every value in JabRefPreferences) in the interface? Or do i just don't understand, what you mean?

Reply:
Yes, you are right. PreferencesServices is the interface for JabRefPreferences which can be injected. Of course you would need to add a lot of getters and setters for the values in the JabRefPreferences class, but I think it's not worth it for the preferences dialog.
But for other dialogs this is useful.

Reply:
What I can do is to inject PreferencesService in PreferencesDialog and downcast it to JabRefPreferences when create the instances of the single tabs.

Reply:
In the long term, it is indeed desirable to inject the preference service. But this requires that all preference options are grouped in certain preference classes (e.g. EntryEditorPreferences). For the moment, it is ok if you pass a `JabRefPreferences` as a constructor argument.